connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -119,12 +119,23 @@ class SparkSession private[sql] (
 
   private def createDataset[T](encoder: AgnosticEncoder[T], data: Iterator[T]): Dataset[T] = {
     newDataset(encoder) { builder =>
-      val localRelationBuilder = builder.getLocalRelationBuilder
-        .setSchema(encoder.schema.json)
       if (data.nonEmpty) {
         val timeZoneId = conf.get("spark.sql.session.timeZone")
-        val arrowData = ConvertToArrow(encoder, data, timeZoneId, allocator)
-        localRelationBuilder.setData(arrowData)
+        val (arrowData, arrowDataSize) = ConvertToArrow(encoder, data, timeZoneId, allocator)
+        if (arrowDataSize <= conf.get(SQLConf.LOCAL_RELATION_CACHE_THRESHOLD.key).toInt) {
```

Comment:
It's kind of weird that we're using an internal API for the client side confs. 

Ideally we leverage the existing stub configs in `connector/connect/common/src/main/scala/org/apache/spark/sql/connect/common/config/ConnectCommon.scala` for now?

Reply:
> It's kind of weird that we're using an internal API for the client side confs.

1. I thought the caching approach could be implemented not only in the connect. 
2. The place you pointed out has constants only but I want to give users some control over the feature.

Reply:
Yes absolutely, your're right this needs to be configurable. My point is mostly that we don't have Spark Confs on the client. In Python we don't have the JVM to parse them on startup for example, you can set them via `spark.conf.set` but that's it. 

My general recommendation would be to avoid pulling in additional SQL/Core dependencies in the client.

Reply:
> My point is mostly that we don't have Spark Confs on the client.

ahh, you meant this:
```suggestion
        if (arrowDataSize <= conf.get("spark.sql.session.localRelationCacheThreshold").toInt) {
```
like @hvanhovell did above:
https://github.com/apache/spark/blob/3814d15bd190bdae69c5c22b461e3f6145cca1f7/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala#L125